File: Android/app/src/main/java/com/mongodb/realm/livedataquickstart/model/ChatMessage.kt
package com.mongodb.realm.livedataquickstart.model

import io.realm.MutableRealmInteger
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.bson.types.ObjectId
import java.time.Instant.now
import java.util.*

open class ChatMessage (rm : String, user: String, msg : String ): RealmObject() {
    @PrimaryKey
    private var _id = ObjectId()
    private var _partition = rm //This is the chat room TODO: change to chat room
    private var author = user
    private var text = msg
    private var timestamp: Date = Date()

    public constructor () : this(rm = "public", user = "anonymous", msg = "test message") {
        _id = ObjectId()
        timestamp = Date()
    }
}

File: iOS/LiveChat/LiveChat/Models/ChatMessage.swift
//
//  ChatMessage.swift
//  LiveChat
//
//  Created by Andrew Morgan on 25/03/2021.
//

import Foundation
import RealmSwift

class ChatMessage: Object, ObjectKeyIdentifiable {
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var room: String
    @Persisted var author: String
    @Persisted var text: String
    @Persisted var timestamp = Date()

    convenience init(author: String, text: String, room: String) {
        self.init()
        self.author = author
        self.text = text
        self.room = room
    }
}
